src/pages/individual/EditIndividualMetadata.jsx

```diff
@@ -123,12 +146,78 @@ export default function EditIndividualMetadata({
           loading={loading}
           display="primary"
           onClick={async () => {
-            const currentPatchableValues = pick(defaultFieldValues, [
-              'names',
-            ]);
+            // validation
+            const requiredFieldErrors = metadata.reduce(
+              (memo, schema) => {
+                if (!schema?.required) return memo;
+
+                const formObject = schema.customField
+                  ? customFieldValues
+                  : defaultFieldValues;
+
+                if (!formObject[schema.name]) {
+                  const intlFieldName = intl.formatMessage({
+                    id: schema.labelId,
+                  });
+                  memo.push(
+                    intl.formatMessage(
+                      { id: 'IS_REQUIRED' },
+                      { field: intlFieldName },
+                    ),
+                  );
+                }
+
+                return memo;
+              },
+              [],
+            );
+
+            setFormErrors(requiredFieldErrors);
+            if (requiredFieldErrors.length > 0) return;
+
+            const names = [];
+            // Always include the defaultName. If it does not already exist, add it.
+            // Otherwise, replace it.
+            const defaultNameGuid = deriveNameGuid(
+              metadata,
+              'defaultName',
+            );
+            const defaultNameProperty = defaultNameGuid
+              ? {
+                  op: 'replace',
+                  value: defaultFieldValues.defaultName,
+                  guid: defaultNameGuid,
+                }
+              : {
+                  op: 'add',
+                  value: defaultFieldValues.defaultName,
+                  context: 'defaultName',
+                };
+            names.push(defaultNameProperty);
+
+            // If there was a nickname, update it with whatever the value is now.
+            // If there was not a nickname, but there is a value now, add it.
+            // If a nickname did not already exist, don't add it.
+            const nicknameGuid = deriveNameGuid(metadata, 'nickname');
+            const nicknameFieldValue = defaultFieldValues.nickname;
+            if (nicknameGuid) {
+              names.push({
+                op: 'replace',
+                value: nicknameFieldValue,
+                guid: nicknameGuid,
+              });
+            } else if (nicknameFieldValue) {
+              names.push({
+                op: 'add',
```

Comment:
i'm pretty sure you can use `op: 'replace'` in either situation and spare the complexity. same with defaultName above

Reply:
I don't think you can. `op: 'replace'` requires a `guid`, which won't exist if the name does not already exist.
https://github.com/WildMeOrg/houston/blob/main/app/modules/individuals/parameters.py#L196-L207

Reply:
you can supply the guid though. but either way is fine